### PR TITLE
Add unit test for testing distributed training with mnist

### LINKED ISSUE #11188: [Stability]Add unit test for distributed training acc with mnist

### PR DESCRIPTION
**Pull request by Yancey0623**
Fixed #11188

### CODE REVIEW

**Comment by panyx0718** [comment]
It seems the timeout issue still happens

**Comment by Yancey1989** [comment]
@panyx0718 yes, but it passes on my develop host, will keep tracing this problem.

File: python/paddle/fluid/tests/unittests/test_dist_mnist.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+import paddle.fluid.profiler as profiler
+from paddle.fluid import core
+import unittest
+from multiprocessing import Process
+import os
+import signal
+
+SEED = 1
+DTYPE = "float32"
+paddle.dataset.mnist.fetch()
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_dist_mnist.py:31]: Use fake data?

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/test_dist_mnist.py (hunk 1 above)
Comment (Yancey0623) [reply python/paddle/fluid/tests/unittests/test_dist_mnist.py:31]: I'm not sure, shall we need to fake all kinds of dataset which we used in the unit test, such as uc_housing, word2vec?

**Comment by typhoonzero** [comment]
Thought @velconia have fixed that SIGKILL problem? Is there still other problems?

File: python/paddle/fluid/tests/unittests/test_dist_mnist.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/test_dist_mnist.py:31]: Team city is mounting cached dataset into docker I think, and using a real dataset is good for testing training to converge

File: python/paddle/fluid/tests/unittests/test_dist_mnist.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                            program=test_program,
+                            feed=feeder.feed(test_data),
+                            fetch_list=[batch_acc, avg_cost])
+                        acc_set.append(float(acc_np))
+                        avg_loss_set.append(float(avg_loss_np))
+                    # get test acc and loss
+                    acc_val = np.array(acc_set).mean()
+                    avg_loss_val = np.array(avg_loss_set).mean()
+                    if float(acc_val
+                             ) > 0.2:  # Smaller value to increase CI speed
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_mnist.py]: can let acc > 0.8 to test model converging.

Comment (Yancey0623) [reply python/paddle/fluid/tests/unittests/test_dist_mnist.py]: Done.

**Review by panyx0718 — APPROVED** [review]

**Review by typhoonzero — COMMENTED** [review]
LGTM

**Pull request merged**